### ISSUE TITLE
Panic when checking non-Setω data definitions with --type-in-type

### ISSUE THREAD

**Issue by iitalics** [agda/agda#3697]
When `--type-in-type` is enabled, data definitions with sort `Set` will panic if a data constructor has an argument with sort `Setω` type:

```agda
{-# OPTIONS --type-in-type #-}

open import Agda.Primitive using (Setω)

data A : Setω where

data B : Set where
  b : A → B

-- Panic: Pattern match failure in do expression at
-- src/full/Agda/TypeChecking/Rules/Data.hs:908:9-14
-- when checking the constructor b in the declaration of B

-- Also occurs if the argument is Setω itself:
data C : Set where
  c : Setω → C

**Comment by andreasabel** [comment]
I think the code responsible for the crash should not even run unless `--cubical`.

**Comment by andreasabel** [comment]
Mmh, apparently cubical Agda needs the `conComp` composition structure for _all_ constructors, even for those data types that were defined when --cubical was off.
Thus, --cubical is not orthogonal; even if I never use --cubical in an Agda development, structures are created just in case.  This consumes resources, and also bears risk that bugs in cubical affect non --cubical developments, as in this case.
To keep --cubical orthogonal, we would need patches to distinguish in our interfaces whether they were compiled with --cubical or not, and recompile with --cubical in case we want to import them from a module using --cubical.

**Comment by Saizan** [comment]
Yeah, the code needs to run, unless we do something fancy with interfaces as you say.

It should be easy to make it more robust, it was written before any of the revamp of universes happened.

**Comment by andreasabel** [comment]
Should this go on 2.6.0.1 !?  It is not a regression, though, as Set\omega is also new in 2.6.0.

**Comment by Saizan** [comment]
I guess it would make sense to include such bugfixes in 2.6.0.1

**Comment by andreasabel** [comment]
You can pick your commit onto the release-2.6.0.1 branch (seems that conflicts will have to be resolved, though).

**Comment by Saizan** [comment]
Cherry-picked the fix into release-2.6.0.1:  423dc1d
